test/zone-spec/proxy.spec.ts

```diff
@@ -0,0 +1,120 @@
+import '../../lib/zone-spec/proxy';
+
+describe('ProxySpec', () => {
+  const ProxyZoneSpec: any = Zone['ProxyZoneSpec'];
+  const delegate = {name: 'delegate'};
+  const proxyZoneSpec = new ProxyZoneSpec(delegate);
+  const proxyZone = Zone.current.fork(proxyZoneSpec);
+
+  describe('properties', () => {
+    it('should expose ProxyZone in the properties', () => {
+      expect(proxyZone.get('ProxyZoneSpec')).toBe(proxyZoneSpec);
+    });
+
+    it('should assert that it is in or out of ProxyZone', () => {
+      expect(() => ProxyZoneSpec.assertPresent()).toThrow();
+      expect(ProxyZoneSpec.isLoaded()).toBe(false);
+      expect(ProxyZoneSpec.get()).toBe(undefined);
+      proxyZone.run(() => {
+        expect(ProxyZoneSpec.isLoaded()).toBe(true);
+        expect(() => ProxyZoneSpec.assertPresent()).not.toThrow();
+      expect(ProxyZoneSpec.get()).toBe(proxyZoneSpec);
+      });
+    });
+
+    it('should reset properties', () => {
+      expect(proxyZone.get('key')).toBe(undefined);
```

Comment:
This test would be more readable if 'key' were a nonsense string like `expect(proxyZone.get('foo'))` - the 'key' is kind of confusing in the documentation so I think it's important to be clear here that it's not a magic string.